haskell/experimental/private/module.bzl

```diff
@@ -133,6 +138,26 @@ def _build_haskell_module(
     src = moduleAttr.src.files.to_list()[0]
     extra_srcs = [f for t in moduleAttr.extra_srcs + ctx.attr.extra_srcs for f in t.files.to_list()]
 
+    user_ghcopts = []
+    user_ghcopts += haskell_library_expand_make_variables("ghcopts", ctx, ctx.attr.ghcopts)
+
+    module_extra_attrs = [
+        [moduleAttr.src],
+        moduleAttr.extra_srcs,
+        moduleAttr.plugins,
+        moduleAttr.tools,
+    ]
+
+    user_ghcopts += expand_make_variables("ghcopts", ctx, moduleAttr.ghcopts, module_extra_attrs)
+
+    if src.extension == "hsc":
+        # TODO[AH] Support version macros
+        hsc_flags, hsc_inputs = preprocess_hsc_flags_and_inputs(dep_info, user_ghcopts, None)
+
+        # TODO[GL] Use idir when providing HaskellReplLoadInfo
```

Comment:
Actually the discussion we had about repls is still valid, but this is also present in `HaskellInfo`, so I should probably propagate it from here in this PR.

Reply:
Tried to address this.

Reply:
I think what I've implemented is broken, but it only shows up when trying to use a `haskell_repl` on a `haskell_library` that has `modules`. I should add a test case for this and fix it.

Reply:
In order to test this with a repl, we need to support repls for targets with `haskell_module`s in them. I'm doing this in a separate PR. So when I have a draft for that I'll probably drop this commit and implement it later.